components/web_discovery/browser/privacy_guard.cc

```diff
@@ -0,0 +1,258 @@
+  if (url.has_username() || url.has_password()) {
+    VLOG(1) << "Ignoring URL due to inclusion of credentials";
+    return true;
+  }
+  if (url.has_port() && url.port_piece() != "443") {
+    VLOG(1) << "Ignoring URL due to non-standard port";
+    return true;
+  }
+  if (matching_url_details && matching_url_details->is_search_engine) {
+    if (url.has_ref() && url.ref_piece().length() > kMaxSearchEngineRefLength) {
+      VLOG(1) << "Ignoring search engine URL due to long ref";
+      return true;
+    }
+  }
+  auto host_piece = url.host_piece();
+  if (host_piece.ends_with(kOnionSiteSuffix) ||
+      host_piece.ends_with(kLocalDomainSuffix) || host_piece == kLocalhost) {
+    VLOG(1) << "Ignoring URL due a local host or onion site";
+    return true;
+  }
+  if (IsPrivateDomainLikely(url.host_piece())) {
+    VLOG(1) << "Ignoring URL due likely private domain";
+    return true;
+  }
+  return false;
+}
+
+bool IsPrivateQueryLikely(const std::string& query) {
+  if (query.length() > kMaxQueryLength) {
+    VLOG(1) << "Ignoring query due to long length";
+    return true;
+  }
+  auto split =
+      base::SplitString(query, " ", base::WhitespaceHandling::KEEP_WHITESPACE,
+                        base::SPLIT_WANT_NONEMPTY);
+  if (split.size() > kMaxQuerySplitLength) {
+    VLOG(1) << "Ignoring query due to long split length";
+    return true;
+  }
+  if (g_regex_util.CheckForLongNumber(query, kMaxQueryNumberLength)) {
+    VLOG(1) << "Ignoring query due to long number";
+    return true;
+  }
+  if (g_regex_util.CheckForEmail(query)) {
+    VLOG(1) << "Ignoring query due to inclusion of email";
+    return true;
+  }
+  if (g_regex_util.CheckQueryHTTPCredentials(query)) {
+    VLOG(1) << "Ignoring query due to potential inclusion of HTTP credentials";
+    return true;
+  }
+  for (const auto& word : split) {
+    if (word.length() > kMaxQueryWordLength) {
+      VLOG(1) << "Ignoring query due to long word";
+      return true;
+    }
+  }
+  if (query.length() >= kHashCheckMinimumLength) {
+    if (IsHashLikely(query, kHashCheckThresholdMultiplier)) {
+      VLOG(1) << "Ignoring query due to potential inclusion of hash";
+      return true;
+    }
+  }
+  return false;
+}
+
+GURL GeneratePrivateSearchURL(const GURL& original_url,
+                              const std::string& query,
+                              const PatternsURLDetails& matching_url_details) {
+  url::RawCanonOutputT<char> query_encoded;
+  url::EncodeURIComponent(query, &query_encoded);
+  std::string query_encoded_str(query_encoded.view());
+  base::ReplaceSubstringsAfterOffset(&query_encoded_str, 0, "%20", "+");
+
+  return GURL(
+      base::StrCat({original_url.scheme(), url::kStandardSchemeSeparator,
+                    original_url.host(), "/",
+                    matching_url_details.search_template_prefix.value_or(
+                        kDefaultSearchPrefix),
+                    query_encoded_str}));
+}
+
+bool ShouldDropLongURL(const GURL& url) {
```

Comment:
the name contradicts what happens in the function: i.e. why `CheckForEmail` implies the URL is long?.. same with forbidden keywords